go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -212,10 +209,12 @@ def replace_directive_substitutions(go_mod_content)
             end
         end
 
-        def substitute_all(file, substitutions)
-          substitutions.reduce(file) do |text, (a, b)|
+        def substitute_all(substitutions)
+          body = substitutions.reduce(File.read("go.mod")) do |text, (a, b)|
```

Comment:
Idea: keep passing `original_go_mod` to avoid reading this again? I don't think it's changed.

Reply:
Looked into it, but we call this method twice, the second time with the inverted substitutions. The second time we haven't read the file yet, we can pull another `File.read` into the `update_files` right before calling this and avoid a single file read, but I think I prefer keeping the code simpler over avoiding a single file read. WDYT?